File: boat-common/src/test/java/test/xyz/srclab/common/base/BaseTest.java
package test.xyz.srclab.common.base;

import org.testng.annotations.Test;
import xyz.srclab.common.base.Format;

/**
 * @author sunqian
 */
public class BaseTest {

    @Test
    public void testFormat() {
        System.out.println(
                Format.fastFormat("This is {} {}!", "fast", "format")
        );
        System.out.println(
                Format.printfFormat("This is %s %s!", "printf", "format")
        );
        System.out.println(
                Format.messageFormat("This is {0} {1}!", "message", "format")
        );
    }
}

File: boat-common/src/test/kotlin/test/xyz/srclab/common/base/BaseTest.kt
package test.xyz.srclab.common.base

import xyz.srclab.common.base.Format.Companion.fastFormat
import xyz.srclab.common.base.Format.Companion.messageFormat
import xyz.srclab.common.base.Format.Companion.printfFormat
import kotlin.test.Test

object FormatTest {
    @Test
    fun testFormat() {
        println(
            "This is {} {}!".fastFormat("fast", "format")
        )
        println(
            "This is %s %s!".printfFormat("printf", "format")
        )
        println(
            "This is {0} {1}!".messageFormat("message", "format")
        )
    }
}